kevin-ashton/server.js

```diff
@@ -0,0 +1,186 @@
+app.get('/new-article', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js, if any, is interacting with this particular piece of `server.js`? What part of CRUD, if any, is being enacted/managed by this particular piece of code?
+  // There is no link/route to this corresponding app,get('/new-article') to a method from article.js. This is the R in CRUD, which stands for READ.
+  response.sendFile('new.html', {root: './public'});
+});
+
+
+// REVIEW: Routes for making API calls to use CRUD Operations on our database
+app.get('/articles', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // Article.fetchAll() is the method that is being used to interact with server.js. The R in CRUD is being enacted in this particular piece of code.
+  client.query('SELECT * FROM articles')
+    .then(function(result) {
+      response.send(result.rows);
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.post('/articles', (request, response) => {
+  console.log('post /articles has been triggered!');
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // The method Article.prototype.insertRecord() is being used to interact with the server.js. The C in CRUD is being enacted by this particular piece of code.
+  let SQL = `
+    INSERT INTO articles(title, author, "authorUrl", category, "publishedOn", body)
+    VALUES ($1, $2, $3, $4, $5, $6);
+  `;
+
+  let values = [
+    request.body.title,
+    request.body.author,
+    request.body.authorUrl,
+    request.body.category,
+    request.body.publishedOn,
+    request.body.body
+  ];
+
+  client.query( SQL, values )
+    .then(function() {
+      response.send('insert complete');
+    })
+    .catch(function(err) {
+      console.error(err);
+    });
+});
+
+app.put('/articles/:id', (request, response) => {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  //Article.prototype.updateRecord is being used as the method to connect the interaction between article.js and server.js. That's the U in CRUD, which stands for UPDATE.
+  let SQL = `UPDATE articles SET title="$1", author="$2", authorUrl="$3", category="$4", publishedOn="$5", body="$6" WHERE article_id=$7;`;
```

Comment:
there needs to be no quotes around the $1, $2 etc.
camelCase variables need to be wrapped in quotes
"authorUrl"=$3